java/vector/src/test/java/org/apache/arrow/vector/TestListViewVector.java

```diff
@@ -0,0 +1,605 @@
+        assertEquals(4, sizeBuffer.getInt(2 * BaseRepeatedValueViewVector.SIZE_WIDTH));
+        assertEquals(0, sizeBuffer.getInt(3 * BaseRepeatedValueViewVector.SIZE_WIDTH));
+
+        // check values
+        assertEquals(12, ((BigIntVector) listViewVector.getDataVector()).get(0));
+        assertEquals(-7, ((BigIntVector) listViewVector.getDataVector()).get(1));
+        assertEquals(25, ((BigIntVector) listViewVector.getDataVector()).get(2));
+        assertEquals(0, ((BigIntVector) listViewVector.getDataVector()).get(3));
+        assertEquals(-127, ((BigIntVector) listViewVector.getDataVector()).get(4));
+        assertEquals(127, ((BigIntVector) listViewVector.getDataVector()).get(5));
+        assertEquals(50, ((BigIntVector) listViewVector.getDataVector()).get(6));
+      }
+
+      setNewValues(listViewVector, offSets1, sizes1, listViewVector.getDataVector());
+
+      listViewWriter.setPosition(4);
+      listViewWriter.startList();
+
+      listViewWriter.bigInt().writeBigInt(121);
+      listViewWriter.bigInt().writeBigInt(-71);
+      listViewWriter.bigInt().writeBigInt(251);
+      listViewWriter.endList();
+
+      listViewVector.setValueCount(5);
+
+      // check offset buffer
+      assertEquals(0, offSetBuffer.getInt(0 * BaseRepeatedValueViewVector.OFFSET_WIDTH));
+      assertEquals(3, offSetBuffer.getInt(1 * BaseRepeatedValueViewVector.OFFSET_WIDTH));
+      assertEquals(3, offSetBuffer.getInt(2 * BaseRepeatedValueViewVector.OFFSET_WIDTH));
+      assertEquals(7, offSetBuffer.getInt(3 * BaseRepeatedValueViewVector.OFFSET_WIDTH));
+      assertEquals(7, offSetBuffer.getInt(4 * BaseRepeatedValueViewVector.OFFSET_WIDTH));
+
+      // check size buffer
+      assertEquals(3, sizeBuffer.getInt(0 * BaseRepeatedValueViewVector.SIZE_WIDTH));
+      assertEquals(0, sizeBuffer.getInt(1 * BaseRepeatedValueViewVector.SIZE_WIDTH));
+      assertEquals(4, sizeBuffer.getInt(2 * BaseRepeatedValueViewVector.SIZE_WIDTH));
+      assertEquals(0, sizeBuffer.getInt(3 * BaseRepeatedValueViewVector.SIZE_WIDTH));
+      assertEquals(3, sizeBuffer.getInt(4 * BaseRepeatedValueViewVector.SIZE_WIDTH));
+
+      // check values
+      assertEquals(12, ((BigIntVector) listViewVector.getDataVector()).get(0));
+      assertEquals(-7, ((BigIntVector) listViewVector.getDataVector()).get(1));
+      assertEquals(25, ((BigIntVector) listViewVector.getDataVector()).get(2));
+      assertEquals(0, ((BigIntVector) listViewVector.getDataVector()).get(3));
+      assertEquals(-127, ((BigIntVector) listViewVector.getDataVector()).get(4));
+      assertEquals(127, ((BigIntVector) listViewVector.getDataVector()).get(5));
+      assertEquals(50, ((BigIntVector) listViewVector.getDataVector()).get(6));
+      assertEquals(121, ((BigIntVector) listViewVector.getDataVector()).get(7));
+      assertEquals(-71, ((BigIntVector) listViewVector.getDataVector()).get(8));
+      assertEquals(251, ((BigIntVector) listViewVector.getDataVector()).get(9));
+    }
+  }
```

Comment:
@lidavidm what do you think about this approach in setting buffers directly for listViews?

Reply:
Not a final API yet, but I added a test case to show case how it could be used.

Reply:
Overly complicated I think. We don't need to let people mix the APIs. We just need a `set(offset, length)`. It's up to you to append to the child vector properly.

Reply:
But we should keep the other API as well right? I mean the `UnionListViewWriter` based API.

Reply:
That has to work, yes.

Reply:
I understand your point. But can we avoid users mixing the APIs? Maybe we can document the usage carefully. 
Plus we are not going to test with mix of APIs right? And we should make sure this is used and tested as independent APIs?

Reply:
At the very least, I am not going to accept the proposed API.

The vectors are all intentionally fairly low level. A simple `set` that lets you directly set the offset/length does exactly what it says on the box and has the least surprises.

Reply:
I understand. I will think about this.

Reply:
@lidavidm I have a doubt regarding the lifetime of the vector created. 

When we create a `ListViewVector` using buffers i.e. offset buffer, size buffer, validity buffer and child vector, we shouldn't copy the buffer values to list vector's buffers right? And the user must ensure the externally given buffers live and we shouldn't do copies right?

Reply:
We never copy buffers in these cases, I believe. It's up to the user to figure out things.